src/proxy-worker.ts

```diff
@@ -64,28 +66,38 @@ class Tunnel extends nodeTunnel.Tunnel {
 		req: nodeTunnel.Request,
 	) => {
 		const { uuid, auth } = this.parseRequest(req);
-		try {
-			await dns.lookup(`${uuid}.vpn`);
-			this.logger.info(`connecting to ${host}:${port}`);
-			const socket = await super.connect(port, host, client, req);
-			metrics.inc(Metrics.ActiveTunnels);
-			metrics.inc(Metrics.TotalTunnels);
-			socket.on('close', () => {
-				metrics.dec(Metrics.ActiveTunnels);
-			});
-			return socket;
-		} catch {
-			// The lookup failed so we try to forward to the correct vpn instance instead
+
+		const deviceIsLocal = async () => {
 			try {
+				await dns.lookup(`${uuid}.vpn`);
+				return true;
+			} catch {
+				return false;
+			}
+		};
+
+		try {
+			if (await deviceIsLocal()) {
+				this.logger.info(`connecting to ${host}:${port}`);
+				const socket = await super.connect(port, host, client, req);
+				metrics.inc(Metrics.ActiveTunnels);
+				metrics.inc(Metrics.TotalTunnels);
+				socket.on('close', () => {
+					metrics.dec(Metrics.ActiveTunnels);
+				});
```

Comment:
You could maybe extend this with
```suggestion
				socket.on('close', (hadError) => {
					metrics.dec(Metrics.ActiveTunnels);
					if (hadError) {
						metrics.inc(Metrics.TunnelSocketTransmissionErrors);
					}
				});
```
to get info on the number of sockets that closed due to transmission errors as opposed to ones that closed gracefully

Reply:
Nice idea, I'll add that now!